pkg/operation/botanist/component/etcd/bootstrap.go

```diff
@@ -76,6 +79,7 @@ func NewBootstrapper(c client.Client, namespace string, image string, imageVecto
 type bootstrapper struct {
 	client               client.Client
 	namespace            string
+	config               *config.GardenletConfiguration
```

Comment:
What about only passing the `ETCDConfig` instead of the full configuration here?

Reply:
I think, passing full configuration is fine. I don't want to do the nil check in deployment spec. Plus, as I am passing the struct by reference, I don't think it would be bad